File: tests/fixtures/python/sample.py
"""Sample Python fixture for tree-sitter parsing tests."""


class PythonClass:
    """A sample Python class."""

    def __init__(self, value):
        self.value = value

    def calculate(self, x, y):
        """Calculate something."""
        return x + y


def top_level_func():
    """A top-level function."""
    return "hello"


def another_function():
    """Another function."""
    pass

File: tests/fixtures/rust/sample.rs
/// Sample Rust fixture for tree-sitter parsing tests

pub struct RustStruct {
    pub value: i32,
}

impl RustStruct {
    pub fn get_value(&self) -> i32 {
        self.value
    }
}

pub fn rust_main() {
    let s = RustStruct { value: 42 };
    println!("{}", s.get_value());
}

fn helper_function() {
    // Helper function
}
